File: WPF/NonUI/OneWaySetter.vb
Namespace Common.NonUI

    Public Class OneWaySetter
        Inherits NonUIElement

#Region "Input Property"
        Public Shared ReadOnly InputProperty As DependencyProperty = DependencyProperty.Register("Input", GetType(Object), GetType(OneWaySetter), New PropertyMetadata(Nothing, AddressOf Input_Changed, AddressOf Input_Coerce))

        Private Shared Function Input_Coerce(ByVal D As DependencyObject, ByVal BaseValue As Object) As Object
            'Dim Self = DirectCast(D, OneWaySetter)

            'Dim Value = DirectCast(BaseValue, Object)

            Return BaseValue
        End Function

        Private Shared Sub Input_Changed(ByVal D As DependencyObject, ByVal E As DependencyPropertyChangedEventArgs)
            Dim Self = DirectCast(D, OneWaySetter)

            'Dim OldValue = DirectCast(E.OldValue, Object)
            Dim NewValue = DirectCast(E.NewValue, Object)

            Self.Output = NewValue
        End Sub

        Public Property Input As Object
            Get
                Return DirectCast(Me.GetValue(InputProperty), Object)
            End Get
            Set(ByVal value As Object)
                Me.SetValue(InputProperty, value)
            End Set
        End Property
#End Region

#Region "Output Property"
        Public Shared ReadOnly OutputProperty As DependencyProperty = DependencyProperty.Register("Output", GetType(Object), GetType(OneWaySetter), New PropertyMetadata(Nothing, AddressOf Output_Changed, AddressOf Output_Coerce))

        Private Shared Function Output_Coerce(ByVal D As DependencyObject, ByVal BaseValue As Object) As Object
            'Dim Self = DirectCast(D, OneWaySetter)

            'Dim Value = DirectCast(BaseValue, Object)

            Return BaseValue
        End Function

        Private Shared Async Sub Output_Changed(ByVal D As DependencyObject, ByVal E As DependencyPropertyChangedEventArgs)
            Dim Self = DirectCast(D, OneWaySetter)

            'Dim OldValue = DirectCast(E.OldValue, Object)
            'Dim NewValue = DirectCast(E.NewValue, Object)

            Await Task.Delay(1)

            Self.Output = Self.Input
        End Sub

        Public Property Output As Object
            Get
                Return DirectCast(Me.GetValue(OutputProperty), Object)
            End Get
            Set(ByVal value As Object)
                Me.SetValue(OutputProperty, value)
            End Set
        End Property
#End Region

    End Class

End Namespace

File: WPF/NonUI/OneWaySetter.cs
using System.Threading.Tasks;
using System.Windows;

namespace Ks.Common.NonUI
{
    public class OneWaySetter : NonUIElement
    {
        public static readonly DependencyProperty InputProperty = DependencyProperty.Register("Input", typeof(object), typeof(OneWaySetter), new PropertyMetadata(null, Input_Changed, Input_Coerce));

        private static object Input_Coerce(DependencyObject D, object BaseValue)
        {
            // Dim Self = DirectCast(D, OneWaySetter)

            // Dim Value = DirectCast(BaseValue, Object)

            return BaseValue;
        }

        private static void Input_Changed(DependencyObject D, DependencyPropertyChangedEventArgs E)
        {
            var Self = (OneWaySetter) D;

            // Dim OldValue = DirectCast(E.OldValue, Object)
            var NewValue = (object) E.NewValue;

            Self.Output = NewValue;
        }

        public object Input
        {
            get => (object) this.GetValue(InputProperty);
            set => this.SetValue(InputProperty, value);
        }

        public static readonly DependencyProperty OutputProperty = DependencyProperty.Register("Output", typeof(object), typeof(OneWaySetter), new PropertyMetadata(null, Output_Changed, Output_Coerce));

        private static object Output_Coerce(DependencyObject D, object BaseValue)
        {
            // Dim Self = DirectCast(D, OneWaySetter)

            // Dim Value = DirectCast(BaseValue, Object)

            return BaseValue;
        }

        private static async void Output_Changed(DependencyObject D, DependencyPropertyChangedEventArgs E)
        {
            var Self = (OneWaySetter) D;

            // Dim OldValue = DirectCast(E.OldValue, Object)
            // Dim NewValue = DirectCast(E.NewValue, Object)

            await Task.Delay(1);

            Self.Output = Self.Input;
        }

        public object Output
        {
            get => (object) this.GetValue(OutputProperty);
            set => this.SetValue(OutputProperty, value);
        }
    }
}
